gpt_index/readers/weaviate/client.py

```diff
@@ -0,0 +1,253 @@
+"""Weaviate-specific serializers for LlamaIndex data structures.
+
+Contain conversion to and from dataclasses that LlamaIndex uses.
+
+"""
+
+import json
+from dataclasses import field
+from typing import Any, Dict, List, Optional, cast
+
+from gpt_index.data_structs.data_structs_v2 import Node
+from gpt_index.data_structs.node_v2 import DocumentRelationship
+from gpt_index.readers.weaviate.utils import (
+    get_by_id,
+    parse_get_response,
+    validate_client,
+)
+from gpt_index.utils import get_new_id
+
+
+def _get_common_properties() -> List[Dict]:
```

Comment:
if this is just for Nodes right now, we could merge with _get_properties

not a big deal though

Reply:
revised.